scripts/keep_largest_component.py

```diff
@@ -0,0 +1,87 @@
+"""
+Keep the largest connected component for each atlas mask.
+
+You can use the SCT venv:
+    source ${SCT_DIR}/python/etc/profile.d/conda.sh                                                                                                                                                                                                                                                            (master|…1)
+    conda activate venv_sct
+
+Then, you can run the script:
+    cd <THIS_REPO>
+    python scripts/keep_largest_component.py -path atlas
+
+The modified files are saved to the `atlas_tmp` folder. To compare the original and modified images you can run:
+    for f in atlas_tmp/*[0-9][0-9].nii.gz;do fsleyes ${f/_tmp/} -cm green ${f} -cm blue-lightblue ${f/.nii/_diff.nii} -cm red-yellow;done
+
+Finally, you can copy the modified images to the original directory:
+    cp atlas_tmp/PAM50_atlas_[0-9][0-9].nii.gz atlas/
+    rm -r atlas_tmp
+
+Resolves: https://github.com/spinalcordtoolbox/PAM50/issues/33
+"""
+
+import argparse
+import shutil
+import nibabel as nib
+import numpy as np
+from scipy import  ndimage
+from pathlib import Path
+
+
+def keep_largest_component(data: np.ndarray) -> np.ndarray:
+    """
+    Keep only the largest connected 3D component.
+
+    :arg data: 3D numpy array
+    :return: 3D numpy array with only the largest connected component
+    """
+    labels, num = ndimage.label(data > 0)       # connected component labeling
```

Comment:
Nit; Python has a built-in way to "discard" return values you don't want to keep, similar to `/dev/null` on Linux. You just pass it to the `_` variable.

```suggestion
    labels, _ = ndimage.label(data > 0)       # connected component labeling
```

Reply:
`num` is used on the next line:
 
```python
sizes = ndimage.sum(np.ones_like(data), labels, index=np.arange(1, num+1))
```

Reply:
Oh; good example of why not to do late night reviews I guess. Sorry about that.

Reply:
no worries! 😄